physlr/physlr.py

```diff
@@ -400,14 +400,50 @@ def diameter_of_tree(g, weight=None):
         return (u, v, diameter)
 
     @staticmethod
-    def determine_backbones_of_trees(g):
-        "Determine the backbones of the maximum spanning trees."
+    def detect_junctions_of_tree(gcomponent, messages, min_branch):
+        """"
+        Detect the junctions in the tree, with >= 3 branches larger than min_branch.
+        """
+        candidate_junctions = {node
```

Comment:
```suggestion
        candidate_junctions = [node
```
A list is more efficient than a set, unless you need the features of a set.